<launch>
  <arg name="use_compressed" default="true" />
    <node pkg="tensorrt_yolov8" type="tensorrt_yolov8_node" name="tensorrt_yolov8_node_right" output="screen">
        <param name="input_topic" value="/pylon_camera_node_right_infra/image_rect/flipped/compressed" if="$(arg use_compressed)"/>
        <param name="input_topic" value="/pylon_camera_node_right_infra/image_rect/flipped" unless="$(arg use_compressed)"/>
        <param name="output_topic" value="/bbox_array_right"/>
        <param name="half_precision" value="true"/>
        <param name="does_support_dynamic_batch_size" value="false"/>
        <param name="max_workspace_size" value="2000000000"/> #2GB
        <param name="onnx_model_name" value="yolov8m.onnx"/>
        <param name="normalize" value="true"/>
        <param name="wanted_class_nums" value="17"/>
        <param name="confidence_threshold" value="0.4"/>
        <param name="nms_threshold" value="0.4"/>
    </node>
</launch>
